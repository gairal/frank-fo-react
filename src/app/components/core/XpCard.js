import React from 'react';
import PropTypes from 'prop-types';
import {
  Grid,
  Typography,
  Avatar,
  Chip,
  Icon,
  Card,
  CardContent,
  CardMedia,
} from '@material-ui/core';
import Utils from '@/Utils';

const XpCard = ({
  img,
  name,
  title,
  date_in, // eslint-disable-line camelcase
  date_out, // eslint-disable-line camelcase
  place,
  subHeadline,
}) => {
  const avatar = icon => (
    <Avatar>
      <Icon
        style={{
          fontSize: '19px',
          color: '#fff',
        }}
      >
        {icon}
      </Icon>
    </Avatar>
  );

  return (
    <Card
      elevation={4}
      style={{
        display: 'flex',
        marginBottom: '1rem',
        padding: '1rem',
      }}
    >
      <CardMedia
        image={`/img/${img}`}
        title={name}
        style={{
          flex: 'none',
        }}
        className="card-media"
      />
      <CardContent
        style={{
          width: '100%',
        }}
      >
        <Typography variant="title" component="h2">
          {`${name} | ${title}`}
        </Typography>
        <Grid
          container
          style={{
            textTransform: 'uppercase',
          }}
          className="card-chips"
        >
          <Grid
            item
            xs={12}
            md
            style={{
              flexGrow: 0,
              marginRight: '0.2rem',
              marginBottom: '.5rem',
            }}
          >
            <Chip
              avatar={avatar('date_range')}
              label={Utils.formatDate(date_in, date_out)}
            />
          </Grid>
          <Grid item xs={12} md>
            <Chip avatar={avatar('location_city')} label={place} />
          </Grid>
        </Grid>
        <Typography
          variant="caption"
          component="p"
          style={{
            marginTop: '0.5rem',
          }}
        >
          {subHeadline}
        </Typography>
      </CardContent>
    </Card>
  );
};

XpCard.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  date_in: PropTypes.string.isRequired,
  date_out: PropTypes.string,
  place: PropTypes.string.isRequired,
  subHeadline: PropTypes.string.isRequired,
};

XpCard.defaultProps = {
  date_out: null,
};

export default XpCard;
