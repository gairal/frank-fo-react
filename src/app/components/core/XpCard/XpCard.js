import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography, Avatar, Chip, Icon, Card, CardContent, CardMedia } from '@material-ui/core';
import Utils from '../../../Utils';

const XpCard = ({
  img, name, title, date_in, date_out, place, subHeadline, // eslint-disable-line camelcase
}) => (
  <Card
    elevation={4}
    style={{ display: 'flex', marginBottom: '1rem', padding: '1rem' }}
  >
    <CardMedia
      image={`/img/${img}`}
      title={name}
      style={{ flex: 'none' }}
      className="card-media"
    />
    <CardContent style={{ width: '100%' }}>
      <Typography variant="title" component="h2">
        {name} | {title}
      </Typography>
      <Grid container style={{ textTransform: 'uppercase' }} className="card-chips">
        <Grid item xs={12} md style={{ flexGrow: 0, marginRight: '0.2rem', marginBottom: '.5rem' }}>
          <Chip
            avatar={<Avatar><Icon style={{ fontSize: '19px', color: '#fff' }}>date_range</Icon></Avatar>}
            label={Utils.formatDate(date_in, date_out)}
          />
        </Grid>
        <Grid item xs={12} md>
          <Chip
            avatar={<Avatar><Icon style={{ fontSize: '19px', color: '#fff' }}>location_city</Icon></Avatar>}
            label={place}
          />
        </Grid>
      </Grid>
      <Typography variant="caption" component="p" style={{ marginTop: '0.5rem' }}>
        {subHeadline}
      </Typography>
    </CardContent>
  </Card>
);

XpCard.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  date_in: PropTypes.string.isRequired,
  date_out: PropTypes.string,
  place: PropTypes.string.isRequired,
  subHeadline: PropTypes.string.isRequired,
};

XpCard.defaultProps = {
  date_out: null,
};

export default XpCard;
