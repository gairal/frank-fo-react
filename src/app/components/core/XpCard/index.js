import XpCard from './XpCard';

export default XpCard;
