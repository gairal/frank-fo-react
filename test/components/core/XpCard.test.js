import React from 'react';
import renderer from 'react-test-renderer';
import XpCard from '@/components/core/XpCard';

describe('components > core > XpCard', () => {
  it('Renders correctly default', () => {
    const component = renderer.create(<XpCard img="test" name="test" title="test" date_in="test" place="test" subHeadline="test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
