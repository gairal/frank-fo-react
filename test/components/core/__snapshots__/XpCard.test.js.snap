// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > core > XpCard Renders correctly default 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation4-8 MuiPaper-rounded-3 MuiCard-root-1"
  style={
    Object {
      "display": "flex",
      "marginBottom": "1rem",
      "padding": "1rem",
    }
  }
>
  <div
    className="MuiCardMedia-root-29 card-media"
    style={
      Object {
        "backgroundImage": "url(\\"/img/test\\")",
        "flex": "none",
      }
    }
    title="test"
  />
  <div
    className="MuiCardContent-root-31"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <h2
      className="MuiTypography-root-32 MuiTypography-title-38"
    >
      test | test
    </h2>
    <div
      className="MuiGrid-container-57 card-chips"
      style={
        Object {
          "textTransform": "uppercase",
        }
      }
    >
      <div
        className="MuiGrid-item-58 MuiGrid-grid-xs-12-97 MuiGrid-grid-md-true-113"
        style={
          Object {
            "flexGrow": 0,
            "marginBottom": ".5rem",
            "marginRight": "0.2rem",
          }
        }
      >
        <div
          className="MuiChip-root-154"
          onKeyDown={[Function]}
          role="button"
          tabIndex={-1}
        >
          <div
            className="MuiAvatar-root-171 MuiAvatar-colorDefault-172 MuiChip-avatar-163"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-174 MuiChip-avatarChildren-166"
              style={
                Object {
                  "color": "#fff",
                  "fontSize": "19px",
                }
              }
            >
              date_range
            </span>
          </div>
          <span
            className="MuiChip-label-167"
          >
            NaN undefined - present
          </span>
        </div>
      </div>
      <div
        className="MuiGrid-item-58 MuiGrid-grid-xs-12-97 MuiGrid-grid-md-true-113"
      >
        <div
          className="MuiChip-root-154"
          onKeyDown={[Function]}
          role="button"
          tabIndex={-1}
        >
          <div
            className="MuiAvatar-root-171 MuiAvatar-colorDefault-172 MuiChip-avatar-163"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-174 MuiChip-avatarChildren-166"
              style={
                Object {
                  "color": "#fff",
                  "fontSize": "19px",
                }
              }
            >
              location_city
            </span>
          </div>
          <span
            className="MuiChip-label-167"
          >
            test
          </span>
        </div>
      </div>
    </div>
    <p
      className="MuiTypography-root-32 MuiTypography-caption-42"
      style={
        Object {
          "marginTop": "0.5rem",
        }
      }
    >
      test
    </p>
  </div>
</div>
`;
